backend/mora/service/address.py

```diff
@@ -375,6 +373,88 @@ def get_one_address(c, addrrel, class_cache=None):
         )
 
 
+class AddressRequest(common.Request):
+
+    __slots__ = *common.Request.__slots__, 'obj_type'
+
+    def __init__(self, *args, **kwargs):
+        self.obj_type = None
+        super().__init__(*args, **kwargs)
+
+    def create(self, req: dict):
+        self.uuid, self.obj_type = get_id_and_type(req)
+
+        # Run through to perform validation
+        get_relation_for(req)
+
+    def edit(self, req: dict):
+        old_entry = util.checked_get(self.request, 'original', {},
+                                     required=True)
+        new_entry = util.checked_get(self.request, 'data', {}, required=True)
+
+        self.uuid, self.obj_type = get_id_and_type(old_entry)
+
+        get_relation_for(old_entry)
+        get_relation_for(new_entry, old_entry)
```

Comment:
I'm not entirely fond of this approach; AFAICT you do everything once, to validate it, and then once more, to issue the request. That seems a bit wasteful and duplicative, to me.

Reply:
Yeah, I see your point. I'll find a suitable way of saving the intermediate result, as I do with the other handlers.